src/formatDate.js

```diff
@@ -50,7 +50,50 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[fromFormat.length - 1]);
+  const newDate = [];
+  const newSeparator = toFormat.slice(-1);
+  let day;
+  let month;
+  let year;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].includes('D')) {
+      day = dateArray[i];
+    }
+
+    if (fromFormat[i].includes('M')) {
+      month = dateArray[i];
+    }
+
+    if (fromFormat[i].includes('Y')) {
+      year = dateArray[i];
+    }
+  }
```

Comment:
istead of create three separate variable suggest to create one object, where key will be format, and value will bie dateArray value.
it will decrease if conditions in next loop